pkg/shared/families/infofinder/sshtopology/scanner.go

```diff
@@ -0,0 +1,364 @@
+	}
+	s.logger.Debugf("Found ssh private keys paths %+v", paths)
+
+	infos, err := s.getFingerprints(paths, types.SSHPrivateKeyFingerprint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ssh private keys fingerprints: %v", err)
+	}
+	s.logger.Debugf("Found ssh private keys fingerprints %+v", infos)
+
+	return infos, nil
+}
+
+func (s *Scanner) getSSHAuthorizedKeysFingerprints(homeUserDir string) ([]types.Info, error) {
+	infos, err := s.getFingerprints([]string{path.Join(homeUserDir, ".ssh/authorized_keys")}, types.SSHAuthorizedKeyFingerprint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ssh authorized keys fingerprints: %v", err)
+	}
+	s.logger.Debugf("Found ssh authorized keys fingerprints %+v", infos)
+
+	return infos, nil
+}
+
+func (s *Scanner) getSSHKnownHostsFingerprints(homeUserDir string) ([]types.Info, error) {
+	infos, err := s.getFingerprints([]string{path.Join(homeUserDir, ".ssh/known_hosts")}, types.SSHKnownHostFingerprint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ssh known hosts fingerprints: %v", err)
+	}
+	s.logger.Debugf("Found ssh known hosts fingerprints %+v", infos)
+
+	return infos, nil
+}
+
+func (s *Scanner) getFingerprints(paths []string, infoType types.InfoType) ([]types.Info, error) {
+	var infos []types.Info
+
+	for _, p := range paths {
+		_, err := os.Stat(p)
+		if os.IsNotExist(err) {
+			s.logger.Debugf("File (%v) does not exist.", p)
+			continue
+		} else if err != nil {
+			return nil, fmt.Errorf("failed to check file: %v", err)
+		}
+
+		var output []byte
+		if output, err = s.executeSSHKeyGenCommand("sha256", p); err != nil {
+			return nil, fmt.Errorf("failed to execute ssh-keygen command: %v", err)
+		}
+
+		infos = append(infos, parseSSHKeyGenCommandOutput(string(output), infoType, p)...)
+	}
+
+	return infos, nil
+}
+
+func (s *Scanner) getPrivateKeysPaths(rootPath string, recursive bool) ([]string, error) {
+	var paths []string
+	err := filepath.Walk(rootPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			if path != rootPath && !recursive {
+				return filepath.SkipDir
+			}
+			return nil
+		}
+
+		isPrivateKeyFile, err := isPrivateKey(path)
+		if err != nil {
+			s.logger.Errorf("failed to verify if file (%v) is private key file - skipping: %v", path, err)
+			return nil
+		}
+
+		if isPrivateKeyFile {
+			paths = append(paths, path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to walks the file tree rooted at %v: %v", rootPath, err)
+	}
+
+	return paths, nil
+}
+
+func isPrivateKey(path string) (bool, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return false, fmt.Errorf("failed to open file: %v", err)
+	}
+	defer f.Close()
+
+	// Splits on newlines by default.
+	scanner := bufio.NewScanner(f)
+
+	for scanner.Scan() {
+		// We only need to look at the first line
+		return strings.Contains(scanner.Text(), "PRIVATE KEY"), nil
```

Comment:
Is it only for PEM private keys?
Do we want to mention it or add todo?

Reply:
e.g. https://superuser.com/questions/1515261/how-to-quickly-identify-ssh-private-key-file-formats